contracts/proxies/CurrencyConvertor.sol

```diff
@@ -0,0 +1,125 @@
+/*
+    Copyright 2021 dYdX Trading Inc.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity ^0.8.7;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import { I_ExchangeWrapper } from "../external/I_ExchangeWrapper.sol";
+import { I_StarkwareContract } from "../external/I_StarkwareContracts.sol";
+
+contract CurrencyConvertor {
+  using SafeERC20 for IERC20;
+
+    // ============ State Variables ============
+
+    I_StarkwareContract public STARKWARE_CONTRACT;
```

Comment:
doesn't seem to be the case that I can do that

Reply:
It is possible, I believe the reason why you can't currently make it `immutable` is because the type is `I_StarkwareContract`. You will need to change the type to `address` to make it compile.

Variables that are `immutable` are much cheaper gas-wise since they are treated similarly to macros (after the contract is deployed and constructor code ran, all `immutable` variables are overwritten in the source code with their actual values). More info on that [here](https://docs.soliditylang.org/en/v0.8.7/contracts.html?highlight=immutable#constant-and-immutable-state-variables). Currently you can only mark [value type variables as `immutable`](https://docs.soliditylang.org/en/v0.8.7/types.html#value-types).

In summary, since this variable is never modified we can save users gas costs by making this variable `immutable`.